modules/spring-config/src/test/java/org/mule/runtime/config/dsl/model/ValueProviderCacheIdGeneratorTestCase.java

```diff
@@ -400,13 +399,24 @@ public void differentConfigsWithSameParameterGetDifferentHash() throws Exception
   }
 
   @Test
-  public void differentValueProviderNameGetsDifferentHash() throws Exception {
+  public void differentValueProviderNameGetsSameHash() throws Exception {
     ArtifactDeclaration app = getBaseApp();
     when(valueProviderModel.requiresConnection()).thenReturn(true);
     when(valueProviderModel.requiresConfiguration()).thenReturn(true);
     Optional<ValueProviderCacheId> opId1 = computeIdFor(app, OPERATION_LOCATION, PROVIDED_PARAMETER_NAME);
     when(valueProviderModel.getProviderName()).thenReturn("newValueProviderName");
     Optional<ValueProviderCacheId> opId2 = computeIdFor(app, OPERATION_LOCATION, PROVIDED_PARAMETER_NAME);
+    checkIdsAreEqual(opId1, opId2);
+  }
+
+  @Test
+  public void differentValueProviderIdGetsDifferentHash() throws Exception {
```

Comment:
We should test that 2 different operations with the same value providerId and same values, return the same hash

Reply:
https://github.com/mulesoft/mule/pull/9467/files#diff-651f29168e45b3db6280ce600b706537R402

Reply:
Ok, I will add another test with this scenario.

Reply:
Done, please review this again.